packages/webapp/src/containers/Finances/ManageCustomExpenseTypes/index.jsx

```diff
@@ -0,0 +1,91 @@
+/*
+ *  Copyright 2023 LiteFarm.org
+ *  This file is part of LiteFarm.
+ *
+ *  LiteFarm is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  LiteFarm is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ *  GNU General Public License for more details, see <https://www.gnu.org/licenses/>.
+ */
+import { useEffect } from 'react';
+import { useDispatch } from 'react-redux';
+import { useTranslation } from 'react-i18next';
+import PureManageCustomTypes from '../../../components/Forms/ManageCustomTypes';
+import { setPersistedPaths } from '../../hooks/useHookFormPersist/hookFormPersistSlice';
+import { icons } from '../NewExpense/ExpenseCategories';
+import labelIconStyles from '../NewExpense/ExpenseCategories/styles.module.scss';
+import useCustomExpenseTypeTileContents from '../useCustomExpenseTypeTileContents';
+
+const addCustomTypePath = '/add_custom_expense';
+
+const getPaths = (typeId) => ({
+  readOnly: `/readOnly_custom_expense/${typeId}`,
+  edit: `/edit_custom_expense/${typeId}`,
+});
+
+export default function ManageExpenseTypes({ history }) {
+  const { t } = useTranslation();
+  const dispatch = useDispatch();
+  const customTypes = useCustomExpenseTypeTileContents();
+
+  const onTileClick = (typeId) => {
+    const { readOnly, edit } = getPaths(typeId);
+    dispatch(setPersistedPaths([readOnly, edit]));
+
+    history.push(readOnly);
+  };
+
+  useEffect(() => {
+    // Manipulate page navigation by pushing "/expense_categories" on top of "/Finances".
+    // When browser's back button or form's back button is clicked, we want to
+    // navigate the user to "/expense_categories" not "/Finances".
+    const unlisten = history.listen(() => {
+      if (history.action === 'POP' && history.location.pathname === '/Finances') {
```

Comment:
For reasons that I don't fully understand, I get the double-back bug _sometimes_ (I think it depends on previous pathing??) after delete or edit only when this `history.location.pathname === '/Finances'` check is included. I don't seem to get it when the check is not included (nor do I see any other bugs) but I wanted to check if you saw the same. Is it absolutely necessary?

Reply:
The double-back bug is seen after completing the form (edit, create, retire).
When the user comes to `/manage_custom_expenses` from `/home` directly for some reason and click the back button after completing the form, we want the user to go to `/home`, not to be forced to go '/expense_categories`. So the check is for that case.
(If we need to fix the double-back bug, I could add some code here, but I would prefer to fix the hook...)

By the way, I see a strange behaviour after editing the type. I'm redirected to the edit view... I'm looking into it.

Reply:
I think I saw that bug too! There was also an extra page refresh. I recorded it because it was so strange but then a reload fixed it!

https://github.com/LiteFarmOrg/LiteFarm/assets/82857964/4fa0dfa2-846b-41b7-b4bf-029ec544fe01

Reply:
It was a form button's submit again that adds "?" to the URL.
I added a type to the button and believe it's working now!

Reply:
Ahhh that pesky submit!!

Reply:
I think it can end up being a **3x** back button situation if you go Add and then Edit the one you just added 😅  But the behaviour is smooth and the routing is otherwise 👌

Reply:
3x back button is quite annoying 😅 Why don't we create a function which works the same as `historyCancel` and always call it when completing forms in sagas instead of `history.push(<the previous page of form's first page>)`?

Reply:
Can it be done?? I have had no success trying to interact with the hookFormPersist stuff from sagas!

Reply:
Sorry 😣 I meant something like this:
 
![Screenshot 2023-09-15 at 2 42 04 PM](https://github.com/LiteFarmOrg/LiteFarm/assets/33141219/6989e460-9627-432f-b5e0-be7e48c8487b)
![Screenshot 2023-09-15 at 2 42 18 PM](https://github.com/LiteFarmOrg/LiteFarm/assets/33141219/85de577a-16a9-4a63-a642-dda552318c64)

This seems to work fine...